stage/stageclr.man
.\" $Id: stageclr.man,v 1.16 2002/10/27 23:29:12 jdurand Exp $
.\"
.\" @(#)$RCSfile: stageclr.man,v $ $Revision: 1.16 $ $Date: 2002/10/27 23:29:12 $ CERN IT-PDP/DM Jean-Philippe Baud
.\" Copyright (C) 1994-2002 by CERN/IT/DS/HSM
.\" All rights reserved
.\"
.TH STAGECLR "1" "$Date: 2002/10/27 23:29:12 $" "CASTOR" "Stage User Commands"
.SH NAME
stageclr \- remove staged files
.SH SYNOPSIS
.B stageclr
[
.BI \-c
] [
.BI \-F
] [
.BI \-h " host"
] [
.BI \-m " minfree"
] [
.BI \-p " pool"
] [
.BI \-\-nodisk
] [
.BI \-remove_from_hsm
] [
.BI \-\-reqid " reqid"
]
.BI \-M " hsmfile"
.P
.B stageclr
[
.BI \-c
] [
.BI \-F
] [
.BI \-h " host"
] [
.BI \-m " minfree"
] [
.BI \-\-nodisk
] [
.BI \-\-reqid " reqid"
]
.BI \-P " pathname"
.P
.B stageclr
[
.BI \-c
] [
.BI \-F
] [
.BI \-h " host"
] [
.BI \-m " minfree"
] [
.BI \-\-nodisk
] [
.BI \-\-reqid " reqid"
]
.BI \-i
.P
.B stageclr
[
.BI \-h " host"
] [
.BI \-F
] [
.BI \-\-nodisk
] [
.BI \-\-reqid " reqid"
]
.BI \-L " link_name"
.SH DESCRIPTION
.B stageclr
removes staged files from disk if the requestor is in the same group as the
user who created the disk file.
.SH OPTIONS
.TP
.BI \-c
conditional clear. If this option is specified the remove operation is done
only if the free space in the pool is lower than a minimum.
This minimum may be specified thru the option
.BR \-m .
.TP
.BI \-F
Forces request to be deleted - admin user only except when used with with \-M option, for which deletion in the HSM is forced and done under requestor uid/gid.
.TP
.BI \-h " host"
Sends the request to the stager on
.IR host .
This may also be specified thru the environment variable STAGE_HOST.
.BI \-i
the list of pathnames to be removed is read from
.IR stdin .
If the operation fails on a path,
.B stageclr
continues with the next path in the list and the global exit code corresponds
to the first failure.
.TP
.BI \-L " link_name"
gives the pathname of the user link to be removed. The data file is not
removed.
.TP
.BI \-M " hsmfile"
clears in the stager catalog the entry corresponding to this HSM file and
remove the staged file from the pool, but it does not remove the file from
the Hierarchical Storage Management System.
This operation should be explicitely requested with the option
.BR \-remove_from_hsm .
.TP
.BI \-m " minfree"
specifies the minimum free space to be available in the pool.
This is a percentage.
For example,
.B \-m 10
corresponds to 10% of free space.
The default minfree value is taken from the configuration file
.B /etc/STGCONFIG.
.TP
.BI \-P " pathname"
gives the path name of the file to be purged,
either full internal name or user link name.
.TP
.BI \-p " poolname"
restricts the action to a given pool.
.TP
.BI \-\-nodisk
Prevent stager daemon to do disk access. The entry is removed from the catalog, but stager do not attempt to remove it physically. Thus space knowledge of the corresponding filesystem is unchanged.
.TP
.BI \-remove_from_hsm
to remove the file from the Hierarchical Storage Management System as well.
.TP
.BI \-\-reqid " reqid"
to remove only entry that has this explicit request ID.
.\" .TP
.\" .B \-\-side "sidenumber"
.\" Tells which side of the device you refer to (applies to multi-side medias files yet in the disk pool, like DVD). Only when used with
.\" .B \-V
.\" option.

.SH RETURN CODES
\
.br
0	Ok.
.br
1	Bad parameter.
.br
2	System error.

.SH SEE ALSO
\fBstage_other_options\fP(1)

.SH AUTHOR
\fBCASTOR\fP Team <castor.support@cern.ch>


CHANGELOG.md
## 5.11.23.0-1 (2026-09-25)

### New Features

- [frontend] Add user configuration improvements for cta-admin-grpc (cta/CTA!1635)

## 5.11.22.0-1 (2026-09-04)

### New Features

- [frontend] Implement JWT blacklisting on the CTA Frontend (cta/CTA!1548)

- [runtime] Add additional validation logic to application config (cta/CTA!1567)


### Bug Fixes

- [scheduler] Fix issues after pg1 release rest: repack admin commands, archive...

- [Tools] Fix 'priority' column in 'cta-admin dr ls' command (cta/CTA!1546)

- [runtime] Make signals, telemetry and health handling more robust (cta/CTA!1554)

- [runtime] Add bounds checking and better error diagnostics to TOML loading (cta/CTA!1559)

## 5.11.21.0-1 (2026-08-25)

### New Features

- [scheduler] Add scheduler workflow metrics to OpenTelementry (cta/CTA!1387)


### Changes

- [Tools] Move cta-admin af ch command to tf ch (cta/CTA!1525)
- [Tools] Replace cta-objectstore-tools.conf with cta-scheduler.conf (cta/CTA!1540)
- [maintd] Change scheduler.objectstore_backend_path to scheduler.config_file (cta/CTA!1518)


### Bug Fixes

- [frontend] Fix logic bug in frontend authentication and SEGFAULT in cta-admin (cta/CTA!1558)
- [Tools] Fix cta-verify-file (cta/CTA!1545)
- [Tools] Add `--storage-class` option to `cta-send-event` (cta/CTA!1542)


### Performance Improvements

- [taped] Skip block reposition request if drive is already positioned correctly (cta/CTA!1527)

## 5.11.20.0-1

### New Features

- [Tools] Add optional storage class statistics, per tape, to cta-admin tool (cta/CTA!1410)
- [Misc] Add event_name to important CTA log messages (cta/CTA!1381)

### Changes

- [Tools] Change short option for cta-admin af ch --idfile from -I to -F (cta/CTA!1453)
- [taped] Taped no longer sets its own user, group and SYS_RAWIO capability (cta/CTA!1426)
- [Misc] Improve .service and .logrotate configuration (cta/CTA!1524)
- [Misc] Modernise RPM spec metadata and file handling (cta/CTA!1507)
- [Misc] Move systemd units to `/usr/lib/systemd/system` (cta/CTA!1497)
- [Misc] Change systemd to weak RPM dependency of CTA (cta/CTA!1509)
- [Misc] Change logrotate to weak RPM dependency of CTA (cta/CTA!1505)
- [Misc] Remove RPM changelog; package CHANGELOG.md as doc (cta/CTA!1508)
- [Misc] Remove noreplace from example configuration files (cta/CTA!1506)
- [taped] Drive process crashing produces CRIT message instead of INFO (cta/CTA!1499)
- [taped] cta-eosdf no longer sources sysconfig (cta/CTA!1455)

### Bug Fixes

- [frontend] Forbid reclaiming tapes with recently deleted files (cta/CTA!1418)
- [frontend] PREPARE gRPC request shouldn't fail without a storage class (cta/CTA!1471)
- [taped] Fix crash due to accidental pointer arithmetic in exception msg (cta/CTA!1502)
- [taped] Fix opentelemetry shutdown race condition with objectstore (cta/CTA!1500)
- [scheduler]: Fix archive unmount decision logic for postgres scheduler (cta/CTA!1450)
- [frontend] Fix frontend catalogue cache for mount rules (cta/CTA!1490)
- [CI] Ensure CTA images trust CERN Root CA (cta/CTA!1535)
- [Misc] Ensure CMake correctly installs executables as PROGRAMS (cta/CTA!1523)

### Security

- [CI] Generate SBOM for each Docker image (cta/CTA!1532)

### Other

- [CI] Add system tests to cover staging tokens on EOS-CTA (cta/CTA!1493)

## 5.11.19.0-1

### Changes

- [frontend] Split Admin/WFE frontends, add mTLS (cta/CTA!1367)
- [catalogue] Forbid adding or changing encryption key of non-empty tapes (cta/CTA!1414)

### Deprecations

- [frontend] Deprecate storage class field in EOS-CTA xattrs (cta/CTA!1424)

### Other

- [CI] Add Enstore and EnstoreLarge tape tests (cta/CTA!1412)

## 5.11.18.0-2

### Bug Fixes

- [Misc] Update required runtime version of opentelemetry-cpp to v1.26.0 (cta/CTA!1370)

## 5.11.18.0-1

### New Features

- [maintd] Add catalogue and telemetry conf files to runtime directory (cta/CTA!1303)
- [Misc] Package basic JSON logging schema (cta/CTA!1304)
- [Misc] Add source location and CTA version to log lines (cta/CTA!1348)

### Changes

- [frontend] Allow admin commands or physics workflows to be disabled on CTA frontend (cta/CTA!1336)
- [Misc] Replace reflect-cpp library with custom implementation (cta/CTA!1291)
- [Misc] Change Catalogue backend install behaviour (cta/CTA!1295)

### Bug Fixes

- [catalogue] Improve CTA resilience to catalogue DB outages during data fetching (cta/CTA!1334)
- [taped] Improve readExactBlock error message contents (cta/CTA!1311)
- [taped] Enabling reading of tapes with misformed OSM tape format (cta/CTA!1343)
- [frontend] Prevent 'cta-admin dr rm' command from crashing the cta-frontend (cta/CTA!1305)
- [frontend] Remove race conditions in NegotiationRequestHandler (cta/CTA!1310)
- [frontend] Fix broken cta-admin fr ls command (cta/CTA!1340)
- [Misc] Fix compile issues when building CTA in Release mode (cta/CTA!1207)

### Performance Improvements

- [CI] Use python xrootd persistent clients for archive instead of xrdcp --parallel (cta/CTA!1274)

### Other

- [Misc] Update opentelemetry-cpp to v1.26.0 (cta/CTA!1321)


## 5.11.17.0-1

### New Features

- [maintd] Add CLI option to use runtime directory for app metadata (cta/CTA!1266)

### Changes

- [maintd] Update maintd to use TOML configuration (cta/CTA!1246)
- [maintd] Use declarative config to initialise OpenTelemetry SDK (cta/CTA!1248)
- [maintd] Add use_unix_domain_socket configuration entry for health server (cta/CTA!1282)
- [maintd] Remove daemon_user and daemon_group settings (cta/CTA!1252)

### Bug Fixes

- [frontend] Fix crash in 'cta-admin dr ls' command when using multiple...
- [frontend] Add error message when trying to change media type of non-empty tape (cta/CTA!1253)

### Removals

- [frontend] Stop using TAPE_POOL.SUPPLY column (cta/CTA!1245)

### Performance Improvements

- [taped] Optimise DriveState queries (cta/CTA!1198)

### Other

- [Misc] Rename *.conf.example files to *.example.conf (cta/CTA!1279)

## 5.11.16.1-1

### Bug Fixes


stage/stageclr.c
/*
 * $Id: stageclr.c,v 1.29 2002/04/11 10:32:38 jdurand Exp $
 */

/*
 * Copyright (C) 1993-1999 by CERN/IT/PDP/DM
 * All rights reserved
 */

#ifndef lint
static char sccsid[] = "@(#)$RCSfile: stageclr.c,v $ $Revision: 1.29 $ $Date: 2002/04/11 10:32:38 $ CERN IT-PDP/DM Jean-Philippe Baud Jean-Damien Durand";
#endif /* not lint */

#include <errno.h>
#include <stdio.h>
#include <signal.h>
#include <sys/types.h>
#include <fcntl.h>
#include <grp.h>
#include <pwd.h>
#include <string.h>
#if defined(_WIN32)
#include <winsock2.h>
#else
#include <unistd.h>
#include <netinet/in.h>
#endif
#include "marshall.h"
#include "stage_api.h"
#include "Cpwd.h"
#include "Cgrp.h"
#include "Cgetopt.h"
#include "serrno.h"

EXTERN_C int  DLL_DECL  send2stgd_cmd _PROTO((char *, char *, int, int, char *, int));  /* Command-line version */
extern int getlist_of_vid _PROTO((char *, char[MAXVSN][7], int *));
extern	char	*getenv();
extern	char	*getconfent();

void cleanup _PROTO((int));
void usage _PROTO((char *));

int reqid_flag = 0;
#ifdef STAGER_SIDE_CLIENT_SUPPORT
int side_flag = 0;
#endif

int main(argc, argv)
		 int	argc;
		 char	**argv;
{
	int c, i, n;
	int errflg = 0;
	int Fflag = 0;
	int Gflag = 0;
	uid_t Guid;
	gid_t gid;
	struct group *gr;
	int Iflag = 0;
	char ibuf[CA_MAXHOSTNAMELEN + 1 + MAXPATH];
	int iflag = 0;
	int Lflag = 0;
	int Mflag = 0;
	int msglen;
	int ntries = 0;
	int nstg161 = 0;
	int numvid;
	int Pflag = 0;
	char *p;
	char path[CA_MAXHOSTNAMELEN + 1 + MAXPATH];
	struct passwd *pw;
	char *q;
	char *sbp;
	char *sbpp;
	char sendbuf[REQBUFSZ];
	char *stghost = NULL;
	uid_t uid;
	char vid[MAXVSN][7];
	char *hsm_host;
	char hsm_path[2 + CA_MAXHOSTNAMELEN + 1 + MAXPATH];
#if defined(_WIN32)
	WSADATA wsadata;
#endif
	int attached = 0;
	char *dummy;
	static struct Coptions longopts[] =
	{
		{"conditionnal",       NO_ARGUMENT,        NULL,      'c'},
		{"force",              NO_ARGUMENT,        NULL,      'F'},
		{"grpuser",            NO_ARGUMENT,        NULL,      'G'},
		{"host",               REQUIRED_ARGUMENT,  NULL,      'h'},
		{"external_filename",  REQUIRED_ARGUMENT,  NULL,      'I'},
		{"input",              NO_ARGUMENT,        NULL,      'i'},
		{"link_name",          REQUIRED_ARGUMENT,  NULL,      'L'},
		{"label_type",         REQUIRED_ARGUMENT,  NULL,      'l'},
		{"migration_filename", REQUIRED_ARGUMENT,  NULL,      'M'},
		{"minfree",            REQUIRED_ARGUMENT,  NULL,      'm'},
		{"pathname",           REQUIRED_ARGUMENT,  NULL,      'P'},
		{"poolname",           REQUIRED_ARGUMENT,  NULL,      'p'},
		{"file_sequence",      REQUIRED_ARGUMENT,  NULL,      'q'},
		{"file_range",         REQUIRED_ARGUMENT,  NULL,      'Q'},
		{"r",                  REQUIRED_ARGUMENT,  NULL,      'r'},
		{"reqid",              REQUIRED_ARGUMENT, &reqid_flag,  1},
#ifdef STAGER_SIDE_CLIENT_SUPPORT
		{"side",               REQUIRED_ARGUMENT, &side_flag,   1},
#endif
		{"vid",                REQUIRED_ARGUMENT,  NULL,      'V'},
		{NULL,                 0,                  NULL,        0}
	};

	/* char repbuf[CA_MAXPATHLEN+1]; */

	uid = getuid();
	gid = getgid();
#if defined(_WIN32)
	if (uid < 0 || gid < 0) {
		fprintf (stderr, STG52);
		exit (USERR);
	}
#endif
	numvid = 0;
#if defined(_WIN32)
	if (WSAStartup (MAKEWORD (2, 0), &wsadata)) {
		fprintf (stderr, STG51);
		exit (SYERR);
	}
#endif
	Coptind = 1;
	Copterr = 1;
	while ((c = Cgetopt_long (argc, argv, "cFGh:I:iL:l:M:m:P:p:q:Q:r:V:", longopts, NULL)) != -1) {
		switch (c) {
		case 'F':
			Fflag++;
			break;
		case 'G':
			Gflag++;
			if ((gr = Cgetgrgid (gid)) == NULL) {
				if (errno != ENOENT) fprintf (stderr, STG33, "Cgetgrgid", strerror(errno));
				fprintf (stderr, STG36, gid);
#if defined(_WIN32)
				WSACleanup();
#endif
				exit (SYERR);
			}
			if ((p = getconfent ("GRPUSER", gr->gr_name, 0)) == NULL) {
				fprintf (stderr, STG10, gr->gr_name);
				errflg++;
			} else {
				if ((pw = Cgetpwnam (p)) == NULL) {
					if (errno != ENOENT) fprintf (stderr, STG33, "Cgetpwnam", strerror(errno));
					fprintf (stderr, STG11, p);
					errflg++;
				} else
					Guid = pw->pw_uid;
			}
			break;
		case 'h':
			stghost = Coptarg;
			break;
		case 'I':
			Iflag++;
			break;
		case 'i':
			iflag++;
			break;
		case 'L':
			Lflag = Coptind - 1;
			if ((n = Coptarg - argv[Lflag]) != 0)
				strncpy (path, argv[Lflag], n);
			if ((c = build_linkname (Coptarg, path+n, sizeof(path)-n, STAGECLR)) == SESYSERR) {
#if defined(_WIN32)
				WSACleanup();
#endif
				exit (SYERR);
			} else if (c)
				errflg++;
			break;
		case 'M':
			Mflag++;
			/* Check if the option -M is attached or not */
			if (strstr(argv[Coptind - 1],"-M") == argv[Coptind - 1]) {
				attached = 1;
			}
			/* We want to know if there is no ':' in the string or, if there is such a ':' */
			/* if there is no '/' before (then is will indicate a hostname)                */
			if (! ISCASTOR(Coptarg)) {
				/* We prepend HSM_HOST only for non CASTOR-like files */
				if ((dummy = strchr(Coptarg,':')) == NULL || (dummy != Coptarg && strrchr(dummy,'/') == NULL)) {
					if ((hsm_host = getenv("HSM_HOST")) != NULL) {
						if (hsm_host[0] != '\0') {
							if (attached != 0) {
								strcpy (hsm_path, "-M");
								strcat (hsm_path, hsm_host);
							} else {
								strcpy (hsm_path, hsm_host);
							}
							strcat (hsm_path, ":");
						}
						strcat (hsm_path, Coptarg);
						argv[Coptind - 1] = hsm_path;
					} else if ((hsm_host = getconfent("STG", "HSM_HOST",0)) != NULL) {
						if (hsm_host[0] != '\0') {
							if (attached != 0) {
								strcpy (hsm_path, "-M");
								strcat (hsm_path, hsm_host);
							} else {
								strcpy (hsm_path, hsm_host);
							}
							strcat (hsm_path, ":");
						}
						strcat (hsm_path, Coptarg);
						argv[Coptind - 1] = hsm_path;
					} else {
						fprintf (stderr, STG54);
						errflg++;
					}
				}
			}
			break;
		case 'P':
			Pflag = Coptind - 1;
			if ((n = Coptarg - argv[Pflag]) != 0)
				strncpy (path, argv[Pflag], n);
			if ((c = build_linkname (Coptarg, path+n, sizeof(path)-n, STAGECLR)) == SESYSERR) {
#if defined(_WIN32)
				WSACleanup();
#endif
				exit (SYERR);
			} else if (c)
				errflg++;
			break;
		case 'r':
			if (strcmp (Coptarg, "emove_from_hsm") != 0)
				errflg++;
			break;
		case 'V':
			errflg += getlist_of_vid ("-V", vid, &numvid);
			break;
		case 0:
			/* Long option without short option correspondance */
			break;
		case '?':
			errflg++;
			break;
		default:
			break;
		}
	}
	if (argc > Coptind) {
		fprintf (stderr, STG16);
		errflg++;
	}
	if (numvid == 0 && Iflag == 0 && iflag == 0 && Lflag == 0 && Pflag == 0 && Mflag == 0 && reqid_flag == 0) {
		fprintf (stderr, STG46);
		errflg++;
	}
	if ((iflag != 0) + (Lflag != 0) + (Pflag != 0) + (Mflag != 0) > 1) {
		fprintf (stderr, STG35, "-i, -L, -M", "-P");
		errflg++;
	}

	if (errflg) {
		usage (argv[0]);
#if defined(_WIN32)
		WSACleanup();
#endif
		exit (1);
	}

	/* Build request header */

	sbp = sendbuf;
	marshall_LONG (sbp, STGMAGIC);
	marshall_LONG (sbp, STAGECLR);
	q = sbp;	/* save pointer. The next field will be updated */
	msglen = 3 * LONGSIZE;
	marshall_LONG (sbp, msglen);

	/* Build request body */

	if ((pw = Cgetpwuid (uid)) == NULL) {
		char uidstr[8];
		if (errno != ENOENT) fprintf (stderr, STG33, "Cgetpwuid", strerror(errno));
		sprintf (uidstr, "%d", uid);
		p = uidstr;
		fprintf (stderr, STG11, p);
#if defined(_WIN32)
		WSACleanup();
#endif
		exit (SYERR);
	}
	marshall_STRING (sbp, pw->pw_name);	/* login name */
	if (Gflag) {
		marshall_WORD (sbp, Guid);
	} else {
		marshall_WORD (sbp, uid);
	}
	marshall_WORD (sbp, gid);

#if ! defined(_WIN32)
	signal (SIGHUP, cleanup);
#endif
	signal (SIGINT, cleanup);
#if ! defined(_WIN32)
	signal (SIGQUIT, cleanup);
#endif
	signal (SIGTERM, cleanup);

	if (iflag) {
		int rc = 0;
		marshall_WORD (sbp, argc+2);
		for (i = 0; i < argc; i++)
			marshall_STRING (sbp, argv[i]);
		marshall_STRING (sbp, "-P");
		sbpp = sbp;
		while (fgets (ibuf, sizeof(ibuf), stdin) != NULL) {
			if ((p = strchr (ibuf, '\n')) != NULL) *p = '\0';
			if ((c = build_linkname (ibuf, path, sizeof(path), STAGECLR)) == SESYSERR) {
#if defined(_WIN32)
				WSACleanup();
#endif
				exit (SYERR);
			} else if (c) {
				if (! rc) rc = USERR;
				continue;
			}
			if (sbp + strlen (path) - sendbuf >= sizeof(sendbuf)) {
				fprintf (stderr, STG38);
				if (! rc) rc = USERR;
				continue;
			} else
				marshall_STRING (sbp, path);

			msglen = sbp - sendbuf;
			sbp = q;
			marshall_LONG (sbp, msglen);	/* update length field */

			while (1) {
				c = send2stgd_cmd (stghost, sendbuf, msglen, 1, NULL, 0);
				if (c == 0 || serrno == EINVAL || serrno == EBUSY) break;
				if (serrno == ENOUGHF) break;
				if (serrno == ESTNACT && nstg161++ == 0) fprintf(stderr, STG161);
				if (serrno != ESTNACT && ntries++ > MAXRETRY) break;
				sleep (RETRYI);
			}
			if (serrno == ENOUGHF) break;
			if (c && serrno != EBUSY && ! rc) rc = serrno;
			sbp = sbpp;
		}
#if defined(_WIN32)
		WSACleanup();
#endif
		exit (rc);
	} else {
		marshall_WORD (sbp, argc);
		for (i = 0; i < argc; i++)
			if ((Pflag && i == Pflag) || (Lflag && i == Lflag)) {
				marshall_STRING (sbp, path);
			} else {
				marshall_STRING (sbp, argv[i]);
			}

		msglen = sbp - sendbuf;
		marshall_LONG (q, msglen);	/* update length field */

		while (1) {
			c = send2stgd_cmd (stghost, sendbuf, msglen, 1, NULL, 0);
			if (c == 0 || serrno == EINVAL) break;
			if (serrno == ENOUGHF) {
				c = 0;
				break;
			}
			if (serrno == ESTNACT && nstg161++ == 0) fprintf(stderr, STG161);
			if (serrno != ESTNACT && ntries++ > MAXRETRY) break;
			sleep (RETRYI);
		}
#if defined(_WIN32)
		WSACleanup();
#endif
		exit (c == 0 ? 0 : rc_castor2shift(serrno));
	}
}

void cleanup(sig)
		 int sig;
{
	signal (sig, SIG_IGN);

#if defined(_WIN32)
	WSACleanup();
#endif
	exit (USERR);
}

void usage(cmd)
		 char *cmd;
{
	fprintf (stderr, "usage: %s ", cmd);
#ifdef STAGER_SIDE_CLIENT_SUPPORT
	fprintf (stderr, "%s%s%s%s",
					 "[-c] [-h stage_host] [-F] [-G] [-I external_filename] [-i] [-L link]\n",
					 "[-l label_type] [-M hsmfile] [-m minfree] [-P path] [-p pool]\n",
					 "[-q file_sequence_number] [-Q file_sequence_range]\n",
					 "[-remove_from_hsm] [--reqid reqid] [-V visual_identifier(s)] [--side sidenumber]\n");
#else
	fprintf (stderr, "%s%s%s%s",
					 "[-c] [-h stage_host] [-F] [-G] [-I external_filename] [-i] [-L link]\n",
					 "[-l label_type] [-M hsmfile] [-m minfree] [-P path] [-p pool]\n",
					 "[-q file_sequence_number] [-Q file_sequence_range]\n",
					 "[-remove_from_hsm] [--reqid reqid] [-V visual_identifier(s)]\n");
#endif
}
